src/data_egress/sqs_listener.py

```diff
@@ -241,6 +263,11 @@ def get_dynamo_records(records):
     if transfer_type == S3_TRANSFER_TYPE:
         destination_bucket = record[DYNAMO_DB_ITEM_DESTINATION_BUCKET]
         destination_prefix = record[DYNAMO_DB_ITEM_DESTINATION_PREFIX]
+        if source_prefix != s3_prefix:
```

Comment:
A unit test to cover this new behaviour maybe, again I realise there is not a lot there to build on.